packages/react/src/components/uiComponents.tsx

```diff
@@ -248,20 +189,34 @@ export function UserProfileLink({ children }: PropsWithChildren<UserProfileLinkP
 }
 
 const _UserProfile = withClerk(
-  ({ clerk, ...props }: WithClerkProp<PropsWithChildren<Without<UserProfileProps, 'customPages'>>>) => {
+  ({
+    clerk,
+    component,
+    fallback,
+    ...props
+  }: WithClerkProp<PropsWithChildren<Without<UserProfileProps, 'customPages'>> & FallbackProp>) => {
+    const mountingStatus = useWaitForComponentMount(component);
+    const shouldShowFallback = mountingStatus === 'rendering' || !clerk.loaded;
+
+    const rendererRootProps = {
+      ...(shouldShowFallback && fallback && { style: { display: 'none' } }),
+    };
+
     const { customPages, customPagesPortals } = useUserProfileCustomPages(props.children);
     return (
-      <Portal
+      <ClerkHostRenderer
```

Comment:
Missing the rendering of fallback here for UserProfile

Reply:
✅ https://github.com/clerk/javascript/pull/4723/commits/67e8e2e2b8b89891c386d6df6eeff7ac3d5ac2be